Create integration test suite and migrate appropriate tests from unit test suite #587

erangeles opened the pull request:
Connects to #587

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/softlayer/sl-ember-components/842)
<!-- Reviewable:end -->

notmessenger commented on tests/unit/components/sl-drop-option-test.js:
```diff
@@ -6,48 +5,35 @@ moduleForComponent( 'sl-drop-option', 'Unit | Component | sl drop option', {
-        this.$().attr( 'role' ),
-        'menuitem',
-        'ARIA role is properly set to "menuitem"'
+        component.get( 'data' ),
```
This assertion is duplicated.

notmessenger commented:
Add a unit test for `tagName`

notmessenger commented:
The function of `optionType()` should be unit tested

notmessenger commented on tests/integration/components/sl-drop-option-test.js:
```diff
@@ -0,0 +1,70 @@
+
+test( 'Has expected initial class name', function( assert ) {
+    this.render( hbs`
+        {{#sl-drop-option}}
```
Change template to `{{sl-drop-option}}`

notmessenger commented on tests/integration/components/sl-drop-option-test.js:
```diff
@@ -0,0 +1,70 @@
+    );
+
+    assert.ok(
+        this.$( '>:first-child' ).find( 'li#tagName' ),
```
What is `#tagName`?

notmessenger commented on tests/integration/components/sl-drop-option-test.js:
```diff
@@ -0,0 +1,70 @@
+
+test( 'Has expected aria-role property', function( assert ) {
+    this.render( hbs`
+        {{#sl-drop-option}}
```
Change template to `{{sl-drop-option}}`

notmessenger commented on tests/integration/components/sl-drop-option-test.js:
```diff
@@ -0,0 +1,70 @@
+
+test( 'Option type class value depends on `label` value', function( assert ) {
+    this.render( hbs`
+        {{#sl-drop-option}}
```
Change template to `{{sl-drop-option}}`

notmessenger commented on tests/integration/components/sl-drop-option-test.js:
```diff
@@ -0,0 +1,70 @@
+    );
+
+    this.render( hbs`
+        {{#sl-drop-option label="test"}}
```
Change template to `{{sl-drop-option}}`

notmessenger commented:
Missing integration tests for:

* `label` in template
* `icon` in template
* action in template

notmessenger commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)
<!-- Sent from Reviewable.io -->

notmessenger commented:
Connects to #587

erangeles commented:
Review status: 0 of 2 files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 10 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLG-r1-10)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/integration/components/sl-drop-option-test.js#L10)):</sup>
Done.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 20 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLG-r1-20)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/integration/components/sl-drop-option-test.js#L20)):</sup>
Done.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 27 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLG-r1-27)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/integration/components/sl-drop-option-test.js#L27)):</sup>
Done.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 40 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLG-r1-40)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/integration/components/sl-drop-option-test.js#L40)):</sup>
Done.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 56 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLG-r1-56)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/integration/components/sl-drop-option-test.js#L56)):</sup>
Done.

---

<sup>**[tests/unit/components/sl-drop-option-test.js, line 12 \[r1\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-Jzz_eLsOA-XbB8zuXLF-r1-12)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0caf7c114fbb569ab735011b55e80054f59c3c51/tests/unit/components/sl-drop-option-test.js#L12)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)
<!-- Sent from Reviewable.io -->

erangeles commented:
Added a unit test for tagname, function of `optionType()` test added, added integration test for label and icon but action has been left in unit/component. btw I think comments are better placed inline with this tool for readability sakes/acknowledgement. 
 

---

Review status: 0 of 2 files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)
<!-- Sent from Reviewable.io -->

notmessenger commented:
Action needs to be tested in both places.  The reason for the integration test is to make sure that the `action` property is being properly wired up in https://github.com/softlayer/sl-ember-components/blob/EmberCli-1.13.8/addon/templates/components/sl-drop-option.hbs#L2 from a template perspective.

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 87 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-K-04c09NUNC2I1Lj7c0)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0c3d75bdd6ebef2330bee67d6f0a88fac8d94f66/tests/integration/components/sl-drop-option-test.js#L87)):</sup>
Should also assert that the `label` property becomes the text of the hyperlink

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 89 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-K-03SIpZieSytPVvHko)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0c3d75bdd6ebef2330bee67d6f0a88fac8d94f66/tests/integration/components/sl-drop-option-test.js#L89)):</sup>
Spelling of "renedred"

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)
<!-- Sent from Reviewable.io -->

erangeles commented:
Addressed 2/3 comments. Still working on action integration test, not ready to review.

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 87 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-K-04c09NUNC2I1Lj7c0)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0c3d75bdd6ebef2330bee67d6f0a88fac8d94f66/tests/integration/components/sl-drop-option-test.js#L87)):</sup>
Done.

---

<sup>**[tests/integration/components/sl-drop-option-test.js, line 89 \[r2\]](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842#-K-03SIpZieSytPVvHko)** ([raw file](https://github.com/softlayer/sl-ember-components/blob/0c3d75bdd6ebef2330bee67d6f0a88fac8d94f66/tests/integration/components/sl-drop-option-test.js#L89)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)
<!-- Sent from Reviewable.io -->

notmessenger commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)
<!-- Sent from Reviewable.io -->

erangeles commented:
implemented action integration test with mcarrol's proposed pattern https://github.com/softlayer/sl-ember-components/blob/EmberCli-1.13.8/tests/integration/components/sl-calendar-test.js#L178 . 

---

Review status: 1 of 2 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)
<!-- Sent from Reviewable.io -->

notmessenger commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/softlayer/sl-ember-components/842)
<!-- Sent from Reviewable.io -->
